### DISCUSSION TITLE
cookieParser - i dont see the cookie in the web

### DISCUSSION THREAD

**Discussion by ShonChassel** [community/community#66710] [Programming Help]
### Select Topic Area

Question

### Feature Area

Issues

### Body

export const login = async (req, res, next) => {
  try {
    const user = await User.findOne({ username: req.body.username });
    if (!user) return next(createError(404, "User not found!"));

    const isPasswordCorrect = await bcrypt.compare(
      req.body.password,
      user.password
    );
    if (!isPasswordCorrect)
      return next(createError(400, "Wrong password or username!"));

    const token = jwt.sign(
      { id: user._id, isAdmin: user.isAdmin },
      process.env.JWT
    );

    const { password, isAdmin, ...otherDetails } = user._doc;

    res
      .cookie("access_token", token, { httpOnly: true,})
      .status(200)
      .json({ details: { ...otherDetails }, isAdmin });
  } catch (err) {
    next(err);
  }
};

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @ShonChassel !

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category and include details beyond just code, the Projects and Issues category is solely related to conversations around the GitHub products Issues and Projects.

We’ve moved your post to our new Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Comment by ilker0** [comment]
Did you add ` credentials: 'include'` for request ? @ShonChassel

**Comment by knownothing10** [comment]
👍 Great! It seems like you have provided a login function in JavaScript. This function uses the await keyword to handle asynchronous operations and try...catch blocks to handle errors. It performs the following steps:

It attempts to find a user based on the provided username using User.findOne.
If a user is not found, it returns a 404 error using createError and passes it to the next middleware.
It checks if the provided password matches the user's password using bcrypt.compare.
If the password is not correct, it returns a 400 error using createError and passes it to the next middleware.
It generates a JWT token using jwt.sign, including the user's ID and isAdmin status.
It extracts the user._doc object, excluding the password and isAdmin properties using object destructuring.
It sets a cookie named "access_token" with the generated token, using res.cookie.
It sends a 200 status code and a JSON response containing the user's details and isAdmin status.
Let me know if you need any further assistance! 😊
